### PR TITLE
Add usage of koa-convert for legacy middleware

### PR DESCRIPTION
**Pull request by menems**
#564, #538 

I change the old version by the used of koa-convert,

### CODE REVIEW

**Comment by yanickrochon** [comment]
Perhaps modify the error message thrown for the dropped generator function support and specify that old middleware should use koa-convert, so issues like #564 are better handled in the future (because we won't see the last of this otherwise).

**Comment by menems** [comment]
I'm agree, but how to test if the function have a old signature? check if the first arg is a context object?

**Comment by felixfbecker** [comment]
You cannot check that but you could give a hint in the error message that the user should use koa-convert if he is using a legacy third-party middleware.

**Comment by menems** [comment]
"Support for generators has been removed. Use Promises or wrap your generator with co.wrap. legacy middleware should be convert using koa-convert" ??

**Comment by felixfbecker** [comment]
Maybe change "Use Promises or wrap your generator with co.wrap" to "See the documentation for examples of how to convert old middleware."

**Comment by menems** [comment]
agree, it's shorter

**Comment by menems** [comment]
oups

File: lib/application.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ module.exports = class Application extends Emitter {
     debug('use %s', fn._name || fn.name || '-');
     if (typeof fn !== 'function') throw new TypeError('middleware must be a function!');
     if (isGeneratorFunction(fn)) {
-      throw new TypeError('Support for generators has been removed. Use Promises or wrap your generator with co.wrap');
+      throw new TypeError('Support for generators has been removed. See the documentation for examples of how to convert old middleware');
```
Comment (jonathanong) [inline lib/application.js]: we should provide a link here

**Comment by menems** [comment]
link added

**Comment by jonathanong** [comment]
@menems oh no, we should be linking to the exact documentation for converting. not just the repo. :)

File: test/application/use.js [hunk 2]
```diff
@@ -66,6 +66,6 @@ describe('app.use(fn)', function(){
   it('should throw error for generator', function(){
     const app = new Koa();
 
-    (() => app.use(function *(){})).should.throw('Support for generators has been removed. Use Promises or wrap your generator with co.wrap');
+    (() => app.use(function *(){})).should.throw('Support for generators has been removed. See the documentation for examples of how to convert old middleware https://github.com/koajs/koa');
```
Comment (jonathanong) [inline test/application/use.js]: for the test, you could also just test that a substring exists so you don't have to make these two strings EXACTLY the same

**Comment by menems** [comment]
linking on the readme anchor?

**Comment by jonathanong** [comment]
yeah

**Comment by menems** [comment]
not complet , next time i will create a branch :)

**Comment by menems** [comment]
now I think is good

**Pull request closed without merging**